src/graph_ir/tests/test_low_graph_souffle_emitter.cc

```diff
@@ -0,0 +1,55 @@
+#include "low_graph_ir.h"
+#include "low_graph_souffle_emitter.h"
+
+#include <vector>
+#include <string>
+#include <iostream>
+
+using namespace std;
+
+LowGraph MakeTestGraph() {
+    // Particle 1: ImageDetector
+    //      reads: 
+    //          sensor_data_packet { camera_feed, video_resolution }
+    //          image_detection_model
+    //      writes:
+    //          image_detection_boxes
+    
+    // Particle 2: ImageSelector
+    //     reads:
+    //         image_detection_boxes
+    //         image_selection   // image selection action from user
+    //     writes:
+    //         selected_image_id
+
+    // Particle 1
+    vector<LowGraphAccessPath> image_detector_inputs = {
+        *(new LowGraphAccessPath("sensor_data_packet.camera_feed")),
```

Comment:
In general, this is poor form, as it guarantees a memory leak. In this test, it won't have any serious consequence, as `exit` will be your garbage collector, but it still doesn't look great. I suspect that this would be better expressed by just creating the test data using https://en.cppreference.com/w/cpp/language/list_initialization .

Reply:
done.